Ads: Boosted Post Test

dbspringer opened the pull request:
This is a test to evaluate the users interest in a potential feature to boost the reach of their content using our paid ad network. 10% of users in AB test `boostedPostSurvey` will see a tour pop up on the Posts page asking them if they are interested in increasing their readership.

Original approach was to start the survey after a use published a post, but there wasn't an easy technical way to start the tour *after* a publish without a page reload. Instead I moved it to the Posts page and added the AB test variant.

Original PR (retracted): https://github.com/Automattic/wp-calypso/pull/10806

To test:
1. Enable `boostedPostSurvey` AB test via dev popup (or add ?tour=boostedPostSurvey to the end of the URL).
1. Go to one of your sites and view the posts.

<img width="1379" alt="screen shot 2017-02-01 at 11 40 04 am" src="https://cloud.githubusercontent.com/assets/273708/22523183/4984cfd2-e873-11e6-8a18-9f52cbc8e882.png">

**Survey Screens**
http://jonburke.polldaddy.com/s/boosted-post

<img width="688" alt="screen shot 2017-02-01 at 11 40 32 am" src="https://cloud.githubusercontent.com/assets/273708/22523193/52637cd4-e873-11e6-866b-e173bd688b3b.png">

<img width="705" alt="screen shot 2017-02-01 at 11 40 23 am" src="https://cloud.githubusercontent.com/assets/273708/22523199/55d6a698-e873-11e6-9603-ac9e51d41b23.png">

folletto commented:
Heya! Looking good! :)

I know it's in progress and you're still discussing it, so at this stage take the below as discussion points 

1. In terms of showing the banner, I'd try to show it once the post is published, attached to the "View post" button on the green notice, which I think it's similar to what was happening in the other PR.
2. I'm sure the trigger conditions to show it (to whom) are still being discussed. I'd probably consider showing it the first time a user publishes a post.
3. An idea for the copy:

> Nice work! Do you know you can give it a visibility boost?
> [ Tell me more ] [ No, thanks ]

mcsf commented on client/layout/guided-tours/tours/boosted-post-tour.js:
```diff
@@ -0,0 +1,52 @@
+		<Step name="init"
+			target=".editor-ground-control__publish-combo"
+			arrow="left-top"
+			placement="beside"
```
You'll probably also want `scrollContainer=".sidebar__region"`, and potentially `shouldScrollTo`. See other tours and how steps react when the user scrolls down the page.

mcsf commented on client/layout/guided-tours/tours/boosted-post-tour.js:
```diff
@@ -0,0 +1,52 @@
+			arrow="left-top"
+			placement="beside"
+		>
+			<p>{ translate( 'Good looking post! Interested in giving it a boost?' ) }</p>
```
Should be "Good-looking", with the hyphen.

That's said, I'd run this by Editorial, since I'm not sold on the copy. "Good-looking" suggests subjective evaluation, not to mention aesthetics. My take is that suggesting that "we" (Calypso) can see and judge the contents of the post can easily come across as either condescending or — perhaps for less technically inclined users — as invasive/creepy.

mcsf commented on client/layout/guided-tours/tours/boosted-post-tour.js:
```diff
@@ -0,0 +1,52 @@
+			isEnabled( 'boosted-post' ),
+			isDesktop,
+			// hasUserRegisteredBefore( new Date( '2016-12-15' ) ),
+			// isAbTestInVariant( 'boostedPostTour', 'enabled' )
```
Have you thought of the triggers you'll be using for this tour?

dbspringer commented:
@folletto @mcsf Sorry for the delay, was down a few days with a bug I picked up on the meetup.

I rejiggered the triggering of the Survey Guided Tour a bit since I couldn't find an easy technical way to start the tour after a publish without a page reload. Instead I moved it to the Posts page and added the AB test variant.

mcsf commented on client/layout/guided-tours/tours/boosted-post-tour.js:
```diff
@@ -0,0 +1,49 @@
+			isEnabled( 'boosted-post-survey' ),
+			isDesktop,
+			isAbTestInVariant( 'boostedPostSurvey', 'enabled' )
+		) }
```
The triggers are really important when it comes to Guided Tours, and the framework was built specifically with that in mind. I don't think that merely popping up this tour on navigation to `/posts` is adequate. I suggest reading our extensive documentation, or — since it has very similar requisites — have a look at this [ongoing PR](https://github.com/Automattic/wp-calypso/pull/10812) by @nosolosw.

In order to have the tour start once a post is published, in a nutshell, you need to pinpoint the Redux action that interests you — action type, but other potential data in the action payload. Use the Redux dev tool for this and try publishing posts in different situations. Then you'll add the relevant action type to the GT's action log (`state.ui.actionLog`) and set up the according selector in GT; again, Andrés's PR is very similar.

dbspringer replied:
@mcsf We changed our approach for this away from immediately after a post is published to gaging interest in all of the posts as they view them. Either option is fine for our needs so it makes sense to to go with an option that requires a bit less overhead to implement it.

mcsf commented on client/layout/guided-tours/tours/boosted-post-tour.js:
```diff
@@ -0,0 +1,49 @@
+				<LinkQuit
+					primary
+					target="_blank"
+					href="http://jonburke.polldaddy.com/s/boosted-post">
```
I'm assuming the intention was to keep the scope of this experiment small and move quickly, but are other ways to collect feedback from users? I know this piece of my feedback doesn't concern GT specifically, but a Polldaddy link feels a little too… artisanal?

mcsf replied:
Minor obs.: the link should be HTTPS as a general principle.

dbspringer replied:
Ah, thanks. Updated to HTTPS. 

We're trying to utilize as many of our pre-existing features as possible for this test, are you aware of another option that feels less artisanal?

mcsf commented on client/layout/guided-tours/tours/boosted-post-tour.js:
```diff
@@ -0,0 +1,49 @@
+		) }
+	>
+		<Step name="init" placement="right">
+			<p>{ translate( 'Interested in increasing your readers?' ) }</p>
```
The copy doesn't feel right (what does it mean to "increase a reader"?). Maybe a _readership_ could be increased, but that can sound vague, especially given the lack of context (and context is something that will be improved once this tour has the appropriate triggers worked in). "Interested in boosting this post and reach a wider audience"? Maybe @benhuberman can pitch in? :)

dbspringer replied:
This is already based on some of the feedback from the editorial folks. See: https://editorialp2.wordpress.com/2017/01/25/can-an-editor-please-take/#comment-10480

benhuberman replied:
@dbspringer is there a way for us to go over all the current copy in the flow, just to make sure all the issues we noticed have been addressed? Or is it still what's in [Jon's survey screens](http://jonburke.polldaddy.com/s/boosted-post)?

dbspringer replied:
@benhuberman Yep, all the copy is in the survey linked above and in the screenshots.

benhuberman replied:
Thanks @dbspringer! In that case, I'd like to propose the following changes -- mostly trims + some rewording for clarity / economy, but also more context + more playfulness, since it wasn't really clear from the existing copy that we're just asking them for their feedback (i.e. a favor), not actually offering a new feature.

popup:
> Wouldn't it be nice if your posts reached a wider audience?

[note: this is assuming the tour starts at /posts, not after or related to any one specific post]

Intro copy:
> We'd like to offer people like you a new way to reach more readers. But we need your help -- answering this short survey will help us create more powerful post-boosting magic. 
>
> Boosting a post promotes it in the “Related Posts” area of WordPress.com sites similar to yours. They look like this: [screenshot]

Survey:

> Q.1: Which of your posts would you like to promote? Copy-and-paste a link to it here:

[note: this step feels superfluous -- would it be possible to pre-populate with the URL of their most recent post? Doesn't this step force them out of the survey?]

> Q.2 How much would you consider spending to boost your post? A bigger budget means more eyes on it.

[note: shouldn't there be "$0" or "I wouldn't pay to boost my post" as an option? At no point up to now was it mentioned that this would be a _paid_ feature.]

> Q.3: Rank which of the following you'd most like to increase (you can drag-and-drop the answers):

Completion screen:

> Thanks for taking the survey.
[The final period was missing. I also wonder if we shouldn't add something along the lines of "Stay tuned for updates!" or something else that feels less abrupt -- I'd be a bit annoyed as a user if I'd just taken a survey and have no idea what the next step is (or even if there is one).

Let me know if I can help with this any further / if you have any questions about my suggestions.

cc @mcsf

jonathangilmoreburke replied:
Thanks a lot Ben. I made changes. We feel we need the URL for a couple of reasons. One is we don't know whose taking the survey without at least the site URL. Also one of the challenges we face at scale is if people want to boost content that will be inappropriate on other sites (it happened once in a prior test). So we want to get a sample to see what we are facing. 

It looks like the final page on Polldaddy is hard set. I can't find a way to edit it.

mcsf commented on client/layout/guided-tours/config-elements/link-quit.js:
```diff
@@ -0,0 +1,48 @@
+			</Button>
+		);
+	}
+}
```
Maybe it's just that I can't see due to my lack of time to review this, but what need is this new component satisfying that the rest of the library couldn't?

mcsf replied:
Also, if we _do_ need this, we will need to update the documentation (https://github.com/Automattic/wp-calypso/tree/master/client/layout/guided-tours/docs).

dbspringer replied:
There wasn't a good way to include a button that was also a link and also quit the tour. Either option we want to close the tour after the action.

If you're aware of another way to handle this, I'd appreciate it if you could point me to it.

rickybanister commented:
Why use the Guided Tour UI for this vs the `banner` component, which in my mind is more directly suited for something like this (it was created for promoting paid things to users after all). 

Won't using the Guided Tour UI water down the efficacy of the tour, and increase confusion with which components serve which purposes?

@folletto you helped engineer the `banner`, wouldn't you agree?

folletto commented:
Heya! I agree but... this has been a bit of a long conversation already for a rather simple test... see the thread in #11118 for the latest bits. :)
